glui_clipboard.cpp

```diff
@@ -0,0 +1,303 @@
+{
+	#ifdef GLUI_GLX	
+	if(nullptr==glui_clipboard_x11::d) 
+	return; 	
+	XEvent ev,ev2;
+	glui_clipboard_x11 p;
+	for(int i=XPending(p.d);i-->0;)
+	{
+		XNextEvent(p.d,&ev);
+		if(ev.type!=SelectionRequest)
+		continue;
+		
+		enum{ xa_targetsN=4};
+		static Atom xa_targets[1+xa_targetsN] = 
+		{
+			XInternAtom(p.d,"XA_TARGETS",0),			
+			XInternAtom(p.d,"UTF8_STRING",0),
+			XInternAtom(p.d,"XA_STRING",0),
+			XInternAtom(p.d,"COMPOUND_TEXT",0), //Xming
+			XInternAtom(p.d,"TEXT",0), //Mentioned in ICCCM manual.
+		};
+			
+		Atom xa = ev.xselectionrequest.target;		
+		ev2.xselection.type = SelectionNotify;
+		ev2.xselection.requestor = ev.xselectionrequest.requestor;
+		ev2.xselection.selection = ev.xselectionrequest.selection;
+		ev2.xselection.target = xa;
+		ev2.xselection.property = None; 
+		ev2.xselection.time = ev.xselectionrequest.time;
+				
+		if(xa==xa_targets[0])
+		{
+			XChangeProperty(p.d,
+			ev2.xselection.requestor,
+			ev2.xselection.property=ev.xselectionrequest.property,
+			xa,CHAR_BIT*sizeof(Atom),PropModeReplace,
+			reinterpret_cast<const unsigned char*>(xa_targets+1),xa_targetsN);
+		}
+		else for(int i=1;i<=xa_targetsN;i++) if(xa==xa_targets[i])
+		{
+			for(size_t i=0;i<p.selected.size();i++)
+			if(p.selected[i].first==ev2.xselection.selection)
+			{
+				XChangeProperty(p.d,
+				ev2.xselection.requestor,
+				ev2.xselection.property=ev.xselectionrequest.property,
+				xa,8,PropModeReplace,
+				reinterpret_cast<const unsigned char*>(p.selected[i].second.c_str()),
+				p.selected[i].second.size());
+			}		
+		}
+		
+		if(ev2.xselection.property==None)
+		{
+			#ifndef NDEBUG
+			char *mystery = XGetAtomName(p.d,xa);
+			if(mystery!=nullptr) printf(mystery);
+			XFree(mystery);
+			#endif
+		}
+		
+		XSendEvent(p.d,ev2.xselection.requestor,1,0,&ev2);
+	}	
+	#endif
+}
+
+template<class T>
+static void glui_clipboard_urlencode(GLUI_String &container, T *p)
+{
+	if(p!=nullptr)	
+	for(int len;*p!=L'\0';p++) if((unsigned)*p>127)
+	{
+		char utf8[4],hex[3];
+		#ifndef _WIN32
+		len = 1; utf8[0] = *p; 
+		#else
+		len = WideCharToMultiByte(CP_UTF8,0,p,1,utf8,4,0,0);
+		#endif
+		for(int i=0;i<len;i++)
+		{
+			//GLUI doesn't display Unicode, so treating like a URL
+			//until it does so.
+			container.push_back('%');
+			snprintf(hex,3,"%02x",(unsigned char)utf8[i]); //itoa
+			container.push_back(hex[0]);
+			container.push_back(hex[1]);
+		}
+	}
+	else container.push_back((char)*p);
+}
+
+bool GLUI_Clipboard_Object::get_text(GLUI_String &container)
+{
+	container.clear();
+
+	if(this!=&GLUI_Clipboard)
+	{
+		if(0!=name.compare("CLIPBOARD")) return false;
+	}
+
+	#ifdef _WIN32
+	if(0==OpenClipboard(0))
+	return false;
+	HGLOBAL h = GetClipboardData(CF_UNICODETEXT);
+	if(h==nullptr)
+	return false;
+	glui_clipboard_urlencode(container,(wchar_t*)GlobalLock(h));	
+	GlobalUnlock(h);
+	CloseClipboard(); 
+	#endif
+
+	#ifdef GLUI_GLX	
+	glui_clipboard_x11 p;
+    if(p.w==0) return false;	
```

Comment:
I simply won't use spaces for indenting. When I do the reformat it will all be tabs. You can change it to spaces in the meantime but I don't think it's important. 

This pull-request is incomplete, like it says, because it only demonstrates copy-paste in the EditText control. Clearly copy-paste must be used by all controls. So by it's nature this pull-request is a to-be-continued thing, and so it's not necessary to apply it even, if transience if spaces and things are so important. But if not applied the next pull-request will just be all the more an omnibus.

@nigels One thing I don't understand, is you are acting as if the code here is a release-branch? We can't treat every change as if it's a release.